src/ert/enkf_main.py

```diff
@@ -198,8 +198,11 @@ def sample_prior(
         parameters = list(parameter_configs.keys())
     for parameter in parameters:
         config_node = parameter_configs[parameter]
-        if config_node.forward_init:
+        if config_node.forward_init or (
+            type(config_node) == GenKwConfig and config_node.design
```

Comment:
Think this logic would be simpler if we used something like `source` instead, also for the future, something like:
```python
ParameterSource(StrEnum):
    forward_init = auto()
    sampled = auto()
    design_matrix = auto()
```

Reply:
Good! Will change correspondigly.